File: frontend/src/types/application.ts
export type ApplicationStatus = "draft" | "submitted" | "withdrawn" | "accepted" | "rejected";
export type WeeklyAvailability = "3–5 saat" | "6–8 saat" | "9–12 saat" | "12+ saat";
export type CommitmentPreference = "exploring" | "trial-sprint" | "long-term" | "cofounder-open";
export type CompensationPreference = "volunteer" | "paid" | "equity" | "open-to-discussion";

export interface ApplicationFormValues {
  selectedRole: string;
  motivation: string;
  contribution: string;
  firstSprintProposal: string;
  weeklyAvailability: WeeklyAvailability;
  commitmentPreference: CommitmentPreference;
  compensationPreferences: CompensationPreference[];
  portfolioUrl: string;
}

export interface ProjectApplication extends ApplicationFormValues {
  id: number;
  projectId: number;
  projectTitle: string;
  applicantId: number;
  applicantName: string;
  applicantEmail: string;
  status: ApplicationStatus;
  createdAt: string;
  updatedAt: string | null;
  submittedAt: string | null;
  reviewedAt: string | null;
}

File: backend/FounderMate.Api/Models/Application.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace FounderMate.Api.Models;

public class Application
{
    public int Id { get; set; }

    public int ProjectId { get; set; }

    [ForeignKey("ProjectId")]
    public Project Project { get; set; } = null!;

    public int ApplicantId { get; set; }

    [ForeignKey("ApplicantId")]
    public User Applicant { get; set; } = null!;

    [MaxLength(200)]
    public string? SelectedRole { get; set; }

    [MaxLength(600)]
    public string? Motivation { get; set; }

    [MaxLength(800)]
    public string? Contribution { get; set; }

    [MaxLength(400)]
    public string? FirstSprintProposal { get; set; }

    [MaxLength(50)]
    public string? WeeklyAvailability { get; set; }

    [MaxLength(50)]
    public string? CommitmentPreference { get; set; }

    [MaxLength(500)]
    public string? CompensationPreferences { get; set; }

    [MaxLength(2048)]
    public string? PortfolioUrl { get; set; }

    [Required, MaxLength(20)]
    public string Status { get; set; } = "Draft"; // Draft, Pending, Accepted, Rejected, Withdrawn

    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime? UpdatedAt { get; set; }
    public DateTime? SubmittedAt { get; set; }
    public DateTime? ReviewedAt { get; set; }
}